pandas/_libs/algos.pyx

```diff
@@ -1218,6 +1221,9 @@ def diff_2d(
     elif (out_t is float64_t
           and (diff_t is float32_t or diff_t is int8_t or diff_t is int16_t)):
         raise NotImplementedError
+    elif out_t is int64_t and diff_t is not int64_t:
```

Comment:
is this hit by a test (or indirectly by a dispatch)?

Reply:
no, this should never be reached (like the line above)

Reply:
ahh so these are guards, okie